### DISCUSSION TITLE
Use of info property on SchemaItem?

### DISCUSSION THREAD

**Discussion by spolloni** [sqlalchemy/sqlalchemy#6938] [Usage Questions] [answered]
I was looking for a way to hold crucial column- and table-level information that have nothing to do with the SQL schema per se. I couldn't find any obvious solution from the docs or SO but eventually stumbled on these lines: 
 
https://github.com/sqlalchemy/sqlalchemy/blob/f5b3103ac2798fb2e5d15541b1489d0d13edf4e7/lib/sqlalchemy/sql/schema.py#L141-L150

A couple questions:

- Is this property documented somewhere and I missed it?
- Is it fine to deliberately use and rely on the `info` dict to contain important information?

My use case would be storing information about an external data source (unrelated to the schema) that will be used to make insertions to a `Table`. I am interested in storing info about **how** this external source relates to the model, e.g. for every `Column` I would store the associated column name in the external source.  Does this seem like a proper use case?

**Comment by zzzeek** [comment, marked as answer]
> I was looking for a way to hold crucial column- and table-level information that have nothing to do with the SQL schema per se. I couldn't find any obvious solution from the docs or SO but eventually stumbled on these lines:
> 
> https://github.com/sqlalchemy/sqlalchemy/blob/f5b3103ac2798fb2e5d15541b1489d0d13edf4e7/lib/sqlalchemy/sql/schema.py#L141-L150
> 
> A couple questions:
> 
>     * Is this property documented somewhere and I missed it?

yeah any docstring you see like that will come out in the generated docs, for that one you can see it here:   https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=schemaitem%20info#sqlalchemy.schema.SchemaItem.info

to find that, go to the "search" box and type "SchemaItem.info"

what it doesn't do is point you to the constructor argument that Table / Column have: https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=schemaitem%20info#sqlalchemy.schema.Table.params.info  https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=schemaitem%20info#sqlalchemy.schema.Column.params.info  

> 
>     * Is it fine to deliberately use and rely on the `info` dict to contain important information?

yes that dictionary is for user-facing usage to put whatever you want in there.

> 
> 
> My use case would be storing information about an external data source (unrelated to the schema) that will be used to make insertions to a `Table`. I am interested in storing info about **how** this external source relates to the model, e.g. for every `Column` I would store the associated column name in the external source. Does this seem like a proper use case?

that's what it's for yup

**Reply by spolloni** [reply]
awesome, really appreciate the help!
